embassy-traits/src/delay.rs

```diff
@@ -0,0 +1,8 @@
+use core::future::Future;
+
+pub trait Delay {
+    type DelayFuture<'a>: Future<Output = ()> + 'a;
+
+    fn delay_ms<'a>(&self, millis: u64) -> Self::DelayFuture<'a>;
```

Comment:
Can you make this `Pin<&'a mut Self>`? 

This way it's possible to implement Delay for a hardware timer. handling interrupts